sympy/series/tests/test_order.py

```diff
@@ -446,3 +446,11 @@ def test_issue_15539():
 
 def test_issue_18606():
     assert unchanged(Order, 0)
+
+def test_issue_9917():
+    assert O(x*sin(x)+1,(x,oo)).expr.simplify()== x*sin(x) + 1
+    assert O((x*sin(x))**2 + 1, (x, oo)).expr.simplify() == x**2*sin(x)**2 + 1
```

Comment:
I guess only the first two testcases should be sufficient. 
Moreover, `simplify()` can be replaced with `cancel()`.
Spaces can be added around `+`, `,` and `(` in the 1st testcase as done in the 2nd testcase.
The code should be added around the issues closer to the current issue number rather than towards the end of the file.